### PR TITLE
Use 'default' as a default JSON style

### PR DESCRIPTION
**Pull request by platan**
This solves a bug with JSON + style reported in https://github.com/badges/shields/issues/1348. 
The problem occurs then we request for a `.json` resource with style of an image (e.g. `.svg`) badge. Why? Because `makeBadge` (`lib/make-badge.js`) sets `flat` as a default value of a badge regardless of a format (there is a `flat-template.svg`, but not `flat-template.json`). For request with unknown style (not an image style nor a json style) it was set to `default` in `makeBadgeData` (`lib/badge-data.js`).
We need style (template name) + format to set default value. That is why I removed `isValidStyle` and I kept only one place where the default values are assigned. 
I'm aware that this bug can be resolved in a different manner, so feel free to comment.

### CODE REVIEW

File: lib/make-badge.spec.js [hunk 1]
```diff
@@ -23,6 +23,18 @@ describe('The badge generator', function () {
     makeBadge({ text: ['cached', 'not-cached'], format: 'svg' });
     assert.deepEqual([..._badgeKeyWidthCache.cache.keys()], ['cached']);
   });
+
+  it('should replace unknown json template with "default"', function () {
+    const jsonBadgeWithUnknownStyle = makeBadge({ text: ['name', 'Bob'], format: 'json', template: 'unknown_style' });
+    const jsonBadgeWithDefaultStyle = makeBadge({ text: ['name', 'Bob'], format: 'json', template: 'default' });
+    assert.strictEqual(jsonBadgeWithUnknownStyle, jsonBadgeWithDefaultStyle);
```
Comment (paulmelnikow) [inline lib/make-badge.spec.js]: Could you add a check that this value is a valid JSON string? And similarly check for svg below?

**Comment by paulmelnikow** [comment]
I've just merged #1419. Could you switch this over to Chai?

File: lib/make-badge.spec.js (hunk 1 above)
Comment (platan) [reply lib/make-badge.spec.js]: What would you suggest to use for JSON validation? I've found https://stackoverflow.com/a/20392392/629762 and `isJSON` from https://github.com/chriso/validator.js.

**Comment by platan** [comment]
Switched over to Chai :-)

File: lib/make-badge.spec.js (hunk 1 above)
Comment (platan) [reply lib/make-badge.spec.js]: Or `expect(JSON.parse(jsonBadgeWithUnknownStyle)).to.deep.equal({name: "name", value: "Bob"})`

Comment (paulmelnikow) [reply lib/make-badge.spec.js]: Yea, that sounds great!

Comment (platan) [reply lib/make-badge.spec.js]: Thanks! Done.

**Comment by platan** [comment]
@paulmelnikow Is there anything missing here? Can we merge this?

**Comment by platan** [comment]
@RedSparr0w Could you please take a look at this PR?

**Comment by RedSparr0w** [comment]
:+1: 
All changes look good to me.
Merged!
:smile:

**Pull request merged**